src/components/containers/derived.js

```diff
@@ -18,6 +19,18 @@ const TraceTypeSection = (props, context) => {
     return <Section {...props} />;
   }
 
+  // when TraceTypeSection is used in a connectedToLayout context
+  if (
+    fullData &&
+    fullData.some(
+      t =>
+        props.traceTypes.includes(t._fullInput.type) ||
+        fullData.some(t => props.traceTypes.includes(t.type))
+    )
+  ) {
+    return <Section {...props} />;
```

Comment:
for readability it might be nice to merge these two ifs, for example by assigning the two conditions to nicely-named functions and then `return (case1() || case2()) ? <Section> : null`...